packages/cozy-client/src/models/instance.js

```diff
@@ -0,0 +1,26 @@
+import get from 'lodash/get'
+const GB = 1000 * 1000 * 1000
+const PREMIUM_QUOTA = 50 * GB
+
+const instance = {
+  //if managet url is here, then it's not selfhosted
+  isSelfHosted: instanceInfo => {
+    return get(instanceInfo, 'context.data.attributes.manager_url')
+      ? false
+      : true
+  },
+  arePremiumLinksEnabled: instanceInfo => {
+    return get(instanceInfo, 'context.data.attributes.enable_premium_links')
+      ? true
+      : false
+  },
+  isFreemiumUser: diskUsage => {
```

Comment:
This is the only function here that does not take instanceInfo, is it expected ? Disk usage comes from another request ?

Reply:
Thanks for asking because I didn't remember that I merged context and instance (currently in a hook something like : 
```jsx
const instancePromise = client
          .getStackClient()
          .fetchJSON('GET', '/settings/instance')

        const contextPromise = client
          .getStackClient()
          .fetchJSON('GET', '/settings/context')
        const promises = [instancePromise, contextPromise]

        const [instanceFetched, contextFetched] = await Promise.all(
          promises.map(p => p.catch(e => e))
        )
        if (!(instanceFetched instanceof Error)) {
          setInstance(instanceFetched)
        }
        if (!(contextFetched instanceof Error)) {
          setContext(contextFetched)
        }
```

In fact we have 3 requests : 
`/settings/context`
`/settings/instance`
`/settings/disk-usage`


`seflHosted` and `premiumLinks` are coming from `/context`
`uuid` from `/instance`
`diskusage` from `/disk-usage`


So I'm wondering if : 
- we should create an unique instanceInfo object with context / instance and disk-usage (we can imagine creating a method in SettingsCollection to fetch the three of them) 
- if we need to isolate each of this setting in its own model 

Personally, I prefer to have only one method to call to get the three and pass the fetched information to an unique InstanceModel. WDY?

Reply:
> Personally, I prefer to have only one method to call to get the three and pass the fetched information to an unique InstanceModel. WDY?

Yes I agree, I am a bit worried though because one call seems very benign but 3 network requests is pretty expensive. It is not expected. Could the stack regroup all this info in a single request ? @nono ?

Reply:
Yep having a `GET /settings` returning (disk-usage / contexte / instance / capabilities) can be a great enhancement!

Reply:
Well, we could, but with HTTP/2, I don't think there is a big difference between one call to fetch with the regrouped information, and 3 calls make in parallel.

@edas what is your opinion?

Reply:
Does HTTP2 reduces the concurrent rqust limit the browsers have ?

Reply:
It seems that the concurrent request limit is raised to 100 with HTTP2, see https://stackoverflow.com/questions/36835972/does-the-per-host-connection-limit-is-raised-with-http-2

Reply:
For requests sent in parallel (you do not wait the response from the first to start the second), having one or three requests is about the same. This was also true for HTTP 1.1 with pipelining.